dotstrings/parser.py

```diff
@@ -153,3 +155,45 @@ def loads(contents: str) -> List[DotStringsEntry]:
         strings.append(DotStringsEntry(key, value, comments))
 
     return strings
+
+def load_plist(file_details: Union[BinaryIO, str]) -> List[DotStringsDictEntry]:
+    """Parse the contents of a .stringsdict file from a file pointer.
+
+    :param file_details: The file pointer or a file path
+
+    :returns: A list of `DotStringEntry`s
+    """
+
+    # If it's a file pointer, read in the contents and parse
+    if not isinstance(file_details, str):
+        contents = file_details.read()
+        return loads_plist(contents)
+
+    with open(file_details, "rb") as stringsdict_file:
+        return load_plist(stringsdict_file)
+
+def loads_plist(contents: bytes) -> List[DotStringsDictEntry]:
+    strings_dict = plistlib.loads(contents, fmt=plistlib.FMT_XML)
+
+    if not isinstance(strings_dict, dict):
+        raise Exception(f"stringsdict format is incorrect")
+
+    entries = []
+    for key, entry in strings_dict.items():
+        if not isinstance(entry, dict):
+            raise Exception(f"stringsdict entry format is incorrect")
+
+        if FORMAT_KEY not in entry:
+            raise Exception(f"NSStringLocalizedFormatKey missing in entry")
+
+        format = entry[FORMAT_KEY]
```

Comment:
I think this loader is doing too much parsing. I'd move it into `dot_stringsdict_entry.py`